### PR TITLE
data stuff

### PR DESCRIPTION
**Pull request by maxim-iwanskyj**
asa

### CODE REVIEW

File: training/collate.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import torch
+from transformers import GPT2Tokenizer
+
+def collate_fn(batch):
+    tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
```
Comment (johnx25bd) [inline training/collate.py]: Move this out of the function body

File: training/train.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# 
+# 
+for batch in dataloader:
+    patches = batch['patches']
+    captions = batch['input_ids']
+    # Optionally, you can get the attention masks if needed
+    attention_mask = batch['attention_mask']
+    
+    print("Patches shape:", patches.shape)  # Should be [batch_size, max_num_patches, patch_dim, patch_dim, channels]
```
Comment (johnx25bd) [inline training/train.py]: The last 3 dimensions should be flattened into one.
- One patch is 16x16 pixels, with 3 channels (r,g,b).
- Each patch should be flattened into 3 vectors, each with a length of 256 (i.e. 16 * 16). 
- Then concatenate the r, g and b vectors into one vector of length 768 (256*3). <- that's a flattened patch.

**Review by johnx25bd — CHANGES_REQUESTED** [review]

**Review by samrae7 — APPROVED** [review]
LGTM 👍

File: training/collate.py (hunk 1 above)
Comment (samrae7) [reply training/collate.py]: done 👍

File: training/train.py (hunk 2 above)
Comment (samrae7) [reply training/train.py]: done 👍

**Pull request merged**